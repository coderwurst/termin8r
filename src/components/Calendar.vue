<template>
  <div class="container">
    <div class="header">{{ msg }} - {{ firstDayOfMonth }}</div>
    <div class="main">
      <div class="calendar-header">
        <span>
          <i @click="decreaseMonth">
            <font-awesome-icon icon="chevron-left" />
          </i>
          {{ month + ' - ' + year }}
          <i @click="increaseMonth">
            <font-awesome-icon icon="chevron-right" />
          </i>
        </span>
      </div>
      <ul class="weekdays">
        <li v-for="day in days" v-bind:key="day.id">{{ day }}</li>
      </ul>
      <ul class="dates">
        <li v-for="blank in firstDayOfMonth" v-bind:key="blank.id">&nbsp;</li>
        <li
          v-for="date in daysInMonth"
          :class="{'current-day': date == initialDate &amp;&amp;
            month == initialMonth && year == initialYear}"
          v-bind:key="date.id"
        >
          {{ date }}
          <span></span>
        </li>
      </ul>
    </div>
    <div class="footer">footer</div>
  </div>
</template>

<script>
import axios from 'axios';
import moment from 'moment';

export default {
  name: 'Calendar',
  data() {
    return {
      today: moment(),
      dateContext: moment(),
      days: ['Su', 'Mo', 'Tu', 'We', 'Th', 'Fr', 'Sa'],
      msg: 'this month starts on day: ',
    };
  },
  created() {
    axios.get('http://localhost:8080/api/meeting/list/12345').then(response => {
      this.meetings = response.data;
    });
  },
  computed: {
    year: function computeYear() {
      return this.dateContext.format('Y');
    },
    month: function computeMonth() {
      return this.dateContext.format('MMMM');
    },
    daysInMonth: function computeDaysInMonth() {
      return this.dateContext.daysInMonth();
    },
    currentDate: function computeCurrentDate() {
      return this.dateContext.get('date');
    },
    firstDayOfMonth: function computeFirstDayOfMonth() {
      return moment(this.dateContext)
        .subtract(this.currentDate - 1, 'days')
        .day();
    },
    // persist today's date, when switching dateContexts during Calander navigation
    initialDate: function storeInitialDate() {
      return this.today.get('date');
    },
    initialMonth: function storeInitialMonth() {
      return this.today.format('MMMM');
    },
    initialYear: function storeInitialYear() {
      return this.today.format('Y');
    },
  },
  methods: {
    increaseMonth: function increaseMonth() {
      this.dateContext = moment(this.dateContext).add(1, 'month');
    },
    decreaseMonth: function decreaseMonth() {
      this.dateContext = moment(this.dateContext).subtract(1, 'month');
    },
  },
};
</script>

<style>
.header {
  grid-area: header;
  background-color: lightblue;
}

.main {
  grid-area: main;
  background-color: purple;
}

.calendar-header {
  display: flex;
  justify-content: center;
  align-items: center;
}

.weekdays,
.dates {
  display: grid;
  grid-template-columns: repeat(7, 1fr);
  list-style: none;
}

.dates {
}

.current-day {
  color: #ffffff;
}

.footer {
  grid-area: footer;
  background-color: greenyellow;
}

.container {
  display: grid;
  grid-template-columns: 1fr 3fr 1fr;
  grid-template-rows: auto;
  grid-template-areas:
    'header header header'
    '. main .'
    'footer footer footer';
  width: 100vw;
  height: 100vh;
}
</style>
