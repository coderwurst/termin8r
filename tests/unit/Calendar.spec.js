/* eslint-disable no-undef */
import { shallowMount } from '@vue/test-utils';
import Calendar from '@/components/Calendar.vue';

describe('Calendar', () => {
  test('is a Vue instance', () => {
    const wrapper = shallowMount(Calendar);
    expect(wrapper.isVueInstance()).toBeTruthy();
  });
});

test('renders to match snapshot', () => {
  const wrapper = shallowMount(Calendar);
  expect(wrapper.element).toMatchSnapshot();
});
