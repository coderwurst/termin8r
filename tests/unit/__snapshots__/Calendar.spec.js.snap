// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders to match snapshot 1`] = `
<div
  class="container"
>
  <div
    class="header"
  >
    this month starts on day:  - 6
  </div>
   
  <div
    class="main"
  >
    <div
      class="calendar-header"
    >
      <span>
        <i>
          <font-awesome-icon
            icon="chevron-left"
          />
        </i>
        
        February - 2020
        
        <i>
          <font-awesome-icon
            icon="chevron-right"
          />
        </i>
      </span>
    </div>
     
    <ul
      class="weekdays"
    >
      <li>
        Su
      </li>
      <li>
        Mo
      </li>
      <li>
        Tu
      </li>
      <li>
        We
      </li>
      <li>
        Th
      </li>
      <li>
        Fr
      </li>
      <li>
        Sa
      </li>
    </ul>
     
    <ul
      class="dates"
    >
      <li>
         
      </li>
      <li>
         
      </li>
      <li>
         
      </li>
      <li>
         
      </li>
      <li>
         
      </li>
      <li>
         
      </li>
       
      <li
        class=""
      >
        
        1
        
        <span />
      </li>
      <li
        class=""
      >
        
        2
        
        <span />
      </li>
      <li
        class=""
      >
        
        3
        
        <span />
      </li>
      <li
        class=""
      >
        
        4
        
        <span />
      </li>
      <li
        class=""
      >
        
        5
        
        <span />
      </li>
      <li
        class=""
      >
        
        6
        
        <span />
      </li>
      <li
        class=""
      >
        
        7
        
        <span />
      </li>
      <li
        class=""
      >
        
        8
        
        <span />
      </li>
      <li
        class=""
      >
        
        9
        
        <span />
      </li>
      <li
        class=""
      >
        
        10
        
        <span />
      </li>
      <li
        class=""
      >
        
        11
        
        <span />
      </li>
      <li
        class=""
      >
        
        12
        
        <span />
      </li>
      <li
        class=""
      >
        
        13
        
        <span />
      </li>
      <li
        class=""
      >
        
        14
        
        <span />
      </li>
      <li
        class=""
      >
        
        15
        
        <span />
      </li>
      <li
        class=""
      >
        
        16
        
        <span />
      </li>
      <li
        class=""
      >
        
        17
        
        <span />
      </li>
      <li
        class=""
      >
        
        18
        
        <span />
      </li>
      <li
        class=""
      >
        
        19
        
        <span />
      </li>
      <li
        class=""
      >
        
        20
        
        <span />
      </li>
      <li
        class=""
      >
        
        21
        
        <span />
      </li>
      <li
        class=""
      >
        
        22
        
        <span />
      </li>
      <li
        class=""
      >
        
        23
        
        <span />
      </li>
      <li
        class=""
      >
        
        24
        
        <span />
      </li>
      <li
        class=""
      >
        
        25
        
        <span />
      </li>
      <li
        class=""
      >
        
        26
        
        <span />
      </li>
      <li
        class=""
      >
        
        27
        
        <span />
      </li>
      <li
        class=""
      >
        
        28
        
        <span />
      </li>
      <li
        class="current-day"
      >
        
        29
        
        <span />
      </li>
    </ul>
  </div>
   
  <div
    class="footer"
  >
    footer
  </div>
</div>
`;
